internal/provider/mssql_role_assignment_resource.go

```diff
@@ -138,14 +139,19 @@ func (r *MssqlRoleAssignmentResource) Read(ctx context.Context, req resource.Rea
 func (r *MssqlRoleAssignmentResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
 	var data MssqlRoleAssignmentResourceModel
 
-	// There is no update - all changes require replacement
-
 	// Read Terraform plan data into the model
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
+	_, err := r.ctx.Client.ReadRoleMembership(ctx, data.Id.ValueString())
```

Comment:
Why this change? Role assignment is generally safe :) Better to just make the API safe against failure instead of "precise"

or is this to fix drift?